### PR TITLE
ORC, Hive: Fix column projection when reading ORC files and `orc.force.positional.evolution` is set to `true` on the default configuration

### PR DESCRIPTION
**Pull request by pvary**
After this change (_[HIVE-20129](https://issues.apache.org/jira/browse/HIVE-20129) Revert to position based schema evolution for orc tables_) in Hive the Iceberg ORC column projection is not working as expected.

The effect:
- If the table has 3 columns (customer_id, first_name, last_name)
- And the query requests the last 2 columns (first_name, last_name)
- We will read the first 2 (customer_id, first_name) instead of the requested ones

I think it would be good to turn off this config for our Iceberg ORC readers as we expect it to be false anyway and other systems using Iceberg might need it to be set differently.

The patch contains 3 groups of changes:
- Test infra changes so we can change HMS configuration (this is used for HiveCatalog FileIO creation)
- Actually setting the `orc.force.positional.evolution` to `true` in the tests - this causes `TestHiveIcebergStorageHandlerLocalScan.testColumnSelection()` to fail
- Setting `orc.force.positional.evolution` to `false` in `ORC.ReadBuilder` to override the configuration values to the expected one - tis fixes the test failures

### CODE REVIEW

**Comment by rdblue** [comment]
If I understand correctly, the id-based projection requests fields by name and if the projection type is changed in the Hive conf, that causes the wrong projection. Right? If so, then I agree with the fix to set this in the conf that we use for readers.

File: mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerTestUtils.java [hunk 1]
```diff
@@ -61,6 +62,7 @@ static TestHiveShell shell() {
     TestHiveShell shell = new TestHiveShell();
     shell.setHiveConfValue("hive.notification.event.poll.interval", "-1");
     shell.setHiveConfValue("hive.tez.exec.print.summary", "true");
+    shell.setHiveConfValue(OrcConf.FORCE_POSITIONAL_EVOLUTION.getHiveConfName(), "true");
```
Comment (rdblue) [inline mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerTestUtils.java:65]: Can you add a comment for this that explains why it is set for the Hive tests? I think it could be confusing later that this is explicitly set to true, only to be overridden by ORC when creating a reader.

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java [hunk 2]
```diff
@@ -106,7 +115,9 @@ public void start(int poolSize) {
 
       TServerSocket socket = new TServerSocket(0);
       int port = socket.getServerSocket().getLocalPort();
-      this.hiveConf = newHiveConf(port);
+      initConf(conf, port);
```
Comment (rdblue) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java:118]: Nit: if `initConf` returned the configuration, this could be one line still: `this.hiveConf = initConf(conf, port);`

File: hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreTest.java [hunk 3]
```diff
@@ -38,7 +38,7 @@
   @BeforeClass
   public static void startMetastore() throws Exception {
     HiveMetastoreTest.metastore = new TestHiveMetastore();
-    metastore.start();
+    metastore.start(new HiveConf(HiveMetastoreTest.class));
```
Comment (rdblue) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreTest.java:41]: Is this change needed? `start()` is still defined and uses `HiveMetastoreTest.class`. The only difference is that this doesn't pass a `Configuration` and the parameterless `start` passes `new Configuration()`.

File: mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerTestUtils.java (hunk 1 above)
Comment (pvary) [reply mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerTestUtils.java:67]: Good point!
Added the comment

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java (hunk 2 above)
Comment (pvary) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java:118]: Before the change the configuration was created in `newHiveConf()`. The new method changes values in the conf parameter. The change meant to emphasize that we are not creating a new conf, only changing values in the one got in the parameter list.

Shall we move back to create a new config and initialize that?

File: hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreTest.java (hunk 3 above)
Comment (pvary) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreTest.java]: Yeah... that is something I forgot to revert.
Thanks for noticing!

Removed the change.

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java (hunk 2 above)
Comment (rdblue) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java:118]: I thought it was a necessary change to pass in the conf for some reason. I'm okay either way. I just wanted to point out a way to make fewer changes.

Comment (rdblue) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java:118]: If conf is passed in, I think the change to `initConf` makes sense. I just think the patch could be slightly smaller. Not a blocker, though.

**Review by rdblue — APPROVED** [review]
